autoload/vimteractive.vim

```diff
@@ -14,53 +62,100 @@ endfunction
 
 " Start a vimteractive session
 function! vimteractive#session(terminal_type)
-
     if has('terminal') == 0
         echoerr "Your version of vim is not compiled with +terminal. Cannot use vimteractive"
         return
     endif
 
+	let l:bufnr = bufnr('%')
+	let l:terminal_type = a:terminal_type
 	if a:terminal_type ==# '-auto-'
 		let term_type = get(g:vimteractive_default_shells, &filetype, &filetype)
 		if has_key(g:vimteractive_commands, term_type)
 			let l:terminal = get(g:vimteractive_commands, term_type)
-			let g:current_term_type = term_type
+			let l:terminal_type = term_type
 		else
 			echoerr "Cannot determine terminal commmand for filetype " . &filetype
 			return
 		endif
 	else
 		let l:terminal = get(g:vimteractive_commands, a:terminal_type)
-		let g:current_term_type = a:terminal_type
 	endif
 
-    if g:vimteractive_terminal != '' && g:vimteractive_terminal != l:terminal
-        echoerr "Cannot run: " . l:terminal " Alreading running: " . g:vimteractive_terminal
-        return
+	" If we have already running Vimteractive terminal
+    if has_key(g:vimteractive_terminal, l:bufnr) && s:check_alive(l:bufnr)
+		if g:vimteractive_current_term_type[l:bufnr] == l:terminal
+			if bufwinnr(g:vimteractive_terminal[l:bufnr]) == -1
+				call s:show_term(l:bufnr)
+			else
+				echom "Terminal of type " . l:terminal_type . " already running for this buffer"
+			endif
+		else
+			echoerr "Cannot run: " . l:terminal
```

Comment:
Is this actually the behaviour that we want? I can imagine a use-case where you might want two alternative versions of the same REPL open at once: e.g. if you want to check a 'clean' run of some code, in case odd behaviour is occurring as a result of global side-effects (which python can be particularly bad for). I don't think there's anything in this new setup that requires this restriction.

Reply:
I don't sure. I can find usecase for two terminals, but it would produce strange situations: what should `:Iterm` do, when there IS a terminal and it's hidden: open new one or show old?

Reply:
I would say that ``:Iterm`` always opens a new terminal (and should be viewed as a 'create terminal' command), but if the terminal is hidden, ctrl-S should still send to the old one (and re-open the old one). I don't think this behaviour is inconsistent with the ``show_term`` function